apps/shell/app.js

```diff
@@ -52,17 +55,38 @@ app.set('views', path.join(__dirname, 'views'));
 app.use(process.env.PASSENGER_BASE_URI || '/', router);
 
 // Setup websocket server
-var server = new http.createServer(app);
-var wss = new WebSocket.Server({ noServer: true });
+const server = new http.createServer(app);
+const wss = new WebSocket.Server({ noServer: true });
+
+let host_whitelist = [];
+if (process.env.SSHHOST_WHITELIST){
+  host_whitelist = process.env.SSHHOST_WHITELIST.split(':');
+}
+
+glob.sync(path.join((process.env.OOD_CLUSTERS || '/etc/ood/config/clusters.d'), '*.y*ml'))
+  .map(yml => yaml.safeLoad(fs.readFileSync(yml)))
+  .filter(config => (config.v2 && config.v2.login && config.v2.login.host) && ! (config.v2 && config.v2.metadata && config.v2.metadata.hidden))
+  .forEach((config) => {
+    let host = config.v2.login.host; //Already did checking above
+    let isDefault = config.v2.login.default;
+    if(isDefault){
+      host_whitelist.unshift(host);
+    } else{
+      host_whitelist.push(host);
+    }
+  });
+host_whitelist.push('default');
```

Comment:
There may be an edge case bug here. Since we are adding "default" as a host, if no cluster config is specified and no process.env.DEFAULT_SSHHOST, if the URL specifies /ssh/default/ we will attempt to ssh to the host "default". Instead this should not work.

The solution, which also would be clearer code, is to not add "default" to host_whitelist, since that is not a valid host anyways. 

Rather, if `process.env.DEFAULT_SSHHOST` is specified, that should be included in `host_whitelist` (since it is a valid host to ssh to).

That does mean when /default/ is in the URL you would need to resolve that to the default host before checking if it is in the whitelist.